TransporterAPI/src/main/java/com/springboot/TransporterAPI/Services/TransporterServiceImpl.java

```
@@ -0,0 +1,149 @@
+package com.springboot.TransporterAPI.Services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import com.springboot.TransporterAPI.Dao.TransporterDao;
+import com.springboot.TransporterAPI.Entity.Transporter;
+import com.springboot.TransporterAPI.Model.TransporterResponse;
+
+@Service
+public class TransporterServiceImpl implements TransporterService {
+
+	@Autowired
+	private TransporterDao transporterdao;
+	
+	
+	@Override
+	public TransporterResponse addTransporter(Transporter transporter) {
+		// TODO Auto-generated method stub
+		TransporterResponse transporterResponse = new TransporterResponse();
+		if(transporter.getName()==null) {
+			transporterResponse.setStatus("Error");
+			transporterResponse.setMessage("Enter name");
+			return transporterResponse;
+		}
+		
+		if(transporter.getPhoneNo()==0) {
```

is this validation correct?

---

Add Regular expression for validating phone numbers.

---

It is used only to check whether phone number is passed as a key in request body or not. Problem that can occur is, if user enter 0 as a value, it will not through an error. It will not affect database at all as we are not calling setter method of phone number. I will try to use regular expression or we can create another class which has only 3 member variables name, kyc and approved. If user enters phone number then we can throw exception similar to argument not found.